src/pseudopeople/interface.py

```diff
@@ -341,12 +341,7 @@ def generate_decennial_census(
         The simulated population has no data for this dataset in the
         specified year or state.
     """
-    filters: list[DataFilter] = []
-    if year is not None:
-        filters.append(DataFilter(DATASET_SCHEMAS.census.date_column_name, "==", year))
-    if state is not None:
-        state_column_name = cast(str, DATASET_SCHEMAS.census.state_column_name)
-        filters.append(DataFilter(state_column_name, "==", get_state_abbreviation(state)))
+    filters: list[DataFilter] = get_generate_data_filters(DATASET_SCHEMAS.census, year, state)
```

Comment:
Now that there's a discrete filter-generator function, how reasonable would it be to just generate the filters inside of the `_generate_dataset`? That would be slightly dryer and then filter stuff could be even more encapulated